test/tests/performance/pbs_equiv_classes_perf.py

```diff
@@ -108,12 +112,159 @@ def test_basic(self):
 
         # wait for cycle to finish
         m = self.scheduler.log_match("Leaving Scheduling Cycle", starttime=t,
-                                     max_attempts=120)
+                                     max_attempts=200)
         self.assertTrue(m)
 
         c = self.scheduler.cycles(lastN=1)[0]
         cycle2_time = c.end - c.start
 
         self.logger.info('Cycle 1: %d Cycle 2: %d Cycle time difference: %d' %
                          (cycle1_time, cycle2_time, cycle1_time - cycle2_time))
-        self.assertTrue(cycle1_time > cycle2_time)
+        self.assertTrue(cycle1_time >= cycle2_time)
+
+    def test_server_queue_limit(self):
+        """
+        Test the performance with hard and soft limits
+        on resources
+        """
+
+        # Update sched_config
+        a = {'preempt_prio':
+             'express_queue, normal_jobs, server_softlimits,\
+             queue_softlimits,server_softlimits+queue_softlimits'}
+        self.scheduler.set_sched_config(a)
```

Comment:
You don't need to set preempt_prio to submit jobs with soft limits.  Setting preempt_prio just allows actual preemption to take place.  I don't see you doing that here.  Now testing this feature with preemption isn't necessarily a bad idea.  A suspended job uses a special select spec to make sure it is placed back on the same nodes it was on before.  If a suspended job is not in a class by itself, it could cause problems.  I don't remember if I took care of this case.  Maybe write a functional test for this?